### PR TITLE
Enable `nifgen` tests for gRPC.

### PR DESCRIPTION
**Pull request by tdunkle**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~

This change does not need to be customer-facing.

~- [ ] I've added tests applicable for this pull request~

This change just enables several existing tests.

### What does this Pull Request accomplish?

This is meant to be a cleaner and more up-to-date version of #1838 . That PR has gotten too messy and unmanageable, since it's attempting to handle several upstream code issues that have been detected and either fixed or worked around during the length of time that the PR's been open. Repeated attempts to get it into a clean state have not worked, so I'm hoping that spinning off a new branch can help get the fundamental change submitted more quickly--then we can use other work items to double back and fix individual test failures as needed.

I've created [internal NI bug 2208439](https://ni.visualstudio.com/DevCentral/_workitems/edit/2208439) to address the issues with `attribute_value` separately, and for now I'm keeping those tests local-only.

Tests in the SystemTests class run both ways. Tests in the TestLibrary class only run locally, and tests in the TestGrpc class only run over gRPC.

For tests that can't run over gRPC, there is a comment before each test on why it doesn't run over gRPC.

### List issues fixed by this Pull Request below, if any.

* [Internal NI bug 2160601](https://ni.visualstudio.com/DevCentral/_workitems/edit/2160601)

### What testing has been done?

107 system tests run and pass now instead of 60 as before. Both gRPC and local tests are running.

### CODE REVIEW

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 1]
```diff
@@ -444,6 +322,65 @@ def test_send_software_edge_trigger_script(self, session):
...
+        with nifgen.Session('', '0', False, 'Simulate=1, DriverSetup=Model:5441;BoardType:PXI') as session:
+            coeff_array = [0 for i in range(95)]
+            coeff_array[0] = -1.0
+            coeff_array[2] = 1.0
+            session.configure_custom_fir_filter_coefficients(coeff_array)
+            session.commit()
+            array = session.get_fir_filter_coefficients()
+            assert len(array) == len(coeff_array)
+            assert array == coeff_array
+    '''
```
Comment (ni-jfitzger) [inline src/nifgen/system_tests/test_system_nifgen.py:382]: Why did you move this?
Is there some reason you don't expect it to work even if #891 is resolved?

Comment (ni-jfitzger) [reply src/nifgen/system_tests/test_system_nifgen.py:382]: If so, I think you should add a comment like you did for the other tests.

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 2]
```diff
@@ -456,6 +393,109 @@ def test_channel_format_types(self):
...
+    def test_create_waveform_numpy_array_int16(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.int16)
+        data.fill(256)
+        assert type(session.create_waveform(data)) is int
+
+    # Test doesn't run over gRPC because numpy isn't supported by gRPC.
+    def test_create_waveform_numpy_array_int16(self, session):
+        data = numpy.ndarray(10000, dtype=numpy.int16)
+        data.fill(256)
+        assert type(session.create_waveform(data)) is int
```
Comment (ni-jfitzger) [inline src/nifgen/system_tests/test_system_nifgen.py:412]: Unless I'm mistaken, you have the same test listed twice in a row, here.

File: src/nifgen/system_tests/test_system_nifgen.py (hunk 1 above)
Comment (ni-jfitzger) [reply src/nifgen/system_tests/test_system_nifgen.py:382]: Nevermind, I see you put a modified copy of it in the other class, too.

File: src/nifgen/system_tests/test_system_nifgen.py (hunk 2 above)
Comment (ni-jfitzger) [reply src/nifgen/system_tests/test_system_nifgen.py:412]: Yep, the travis-ci says that it's being redefined: https://app.travis-ci.com/github/ni/nimi-python/jobs/587616933

**Comment by ni-jfitzger** [comment]
I think it would be helpful, if you added your blurb from the previous PR to this PR's description:

```
Tests in the SystemTests class run both ways. Tests in the TestLibrary class only run locally, and tests in the TestGrpc class only run over gRPC.

For tests that can't run over gRPC, there is a comment before each test on why it doesn't run over gRPC.
```

**Review by ni-jfitzger — CHANGES_REQUESTED** [review]
Need to fix the duplicate test issue. Otherwise, this looks okay to me.

File: src/nifgen/system_tests/test_system_nifgen.py (hunk 2 above)
Comment (tdunkle) [reply src/nifgen/system_tests/test_system_nifgen.py]: I've removed the duplicate test definition.

**Review by ni-jfitzger — APPROVED** [review]

**Review by marcoskirsch — APPROVED** [review]
